core/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -47,6 +40,12 @@
  */
 public class RoutingNodes implements Iterable<RoutingNode> {
 
+    private final static Random RANDOM;
+
+    static {
+        RANDOM = Randomness.create();
+    }
+
```

Comment:
and one more alternative, create a helper method called FooBar.shuffle(List) that does this. It should be as safe as the above, since randomizedrunner has the same checks. But the nice thing is contained to one easy static method

```
/** shuffles a list like Collections.shuffle, except during tests, it uses the current RandomizedContext seed */
public static void shuffle(List list) {
  final Random random;
  if (System.getProperty("tests.seed") != null) {
    // running in tests
    try ... Class.forName("com.carrotsearch.randomizedtesting.RandomizedContext");
     ...
    random = invoke("getRandom");
  } else {
    random = new Random();
  }
  Collections.shuffle(list, random);
}
```

http://labs.carrotsearch.com/download/randomizedtesting/2.0.0/api/randomizedtesting-runner/com/carrotsearch/randomizedtesting/RandomizedContext.html#getRandom%28%29